realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.rx;
+
+import io.realm.DynamicRealm;
+import io.realm.DynamicRealmObject;
+import io.realm.Realm;
+import io.realm.RealmChangeListener;
+import io.realm.RealmList;
+import io.realm.RealmObject;
+import io.realm.RealmQuery;
+import io.realm.RealmResults;
+import rx.Observable;
+import rx.Subscriber;
+import rx.functions.Action0;
+import rx.subscriptions.Subscriptions;
+
+/**
+ * Factory class for creating Observables for RxJava (<=1.0.*).
+ *
+ * @see Realm#asObservable()
+ * @see RealmObject#asObservable()
+ * @see RealmResults#asObservable()
+ * @see RealmQuery#asObservable()
+ * @see DynamicRealm#asObservable()
+ * @see DynamicRealmObject#asObservable()
+ */
+public class RealmObservableFactory implements RxObservableFactory {
+
+    private boolean rxJavaAvailble;
+
+    public RealmObservableFactory() {
+        try {
+            Class.forName("rx.Observable");
+            rxJavaAvailble = true;
+        } catch (ClassNotFoundException ignore) {
+            rxJavaAvailble = false;
+        }
+    }
+
+    @Override
+    public Observable<Realm> from(final Realm realm) {
+        checkRxJavaAvailable();
+        return Observable.create(new Observable.OnSubscribe<Realm>() {
+            @Override
+            public void call(final Subscriber<? super Realm> subscriber) {
+                final RealmChangeListener listener = new RealmChangeListener() {
+                    @Override
+                    public void onChange() {
+                        subscriber.onNext(realm);
```

Comment:
I'd still vote for following the contract and adding the check anyways. If some code changes down the road, or the unsubscribe happens on another scheduler, it could end up breaking the contract.